weavetest/assert/assert.go

```diff
@@ -0,0 +1,36 @@
+package assert
+
+import (
+	"reflect"
+	"testing"
+)
+
+// NoErr fails the test if given error is not nil.
+func NoErr(t testing.TB, err error) {
+	t.Helper()
+	if !errIsNil(err) {
+		t.Fatalf("want the error to be nil, got %+v", err)
+	}
+}
+
+// errIsNil returns true if value represented by the given error is nil.
+//
+// Most of the time a simple == check is enough. There is a very narrowed
+// spectrum of cases  where a more sophisticated check is required.
+func errIsNil(err error) bool {
+	if err == nil {
+		return true
+	}
+	if val := reflect.ValueOf(err); val.Kind() == reflect.Ptr {
+		return val.IsNil()
+	}
+	return false
+}
+
+// Equal fails the test if two values are not equal.
+func Equal(t testing.TB, want, got interface{}) {
+	t.Helper()
+	if !reflect.DeepEqual(want, got) {
+		t.Fatalf("values not equal \nwant %v\n got %v", want, got)
```

Comment:
For equal, the naming is perfect, but can you make this `Errorf()`?

Reply:
If we are not consistent with Fatalf it will be hard to use one package only. I can see how this can be very annoying if you don't see all the errors but only one after the other in debugging but on the other side we may run into panics if the dev expected the test to stop at this line.